types/src/multi_signature.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_crypto::bls12381;
+use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
+use std::collections::HashMap;
+
+use move_deps::move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Serialize};
+
+/// This struct represents the aggregated BLS signature representation that contains an aggregated
+/// BLS signature and a bit mask representing the set of validators participating in the signing
+/// process
+#[derive(
+    Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize, CryptoHasher, BCSCryptoHash,
+)]
+pub struct MultiSignature {
+    validator_bitmask: Vec<bool>,
+    multi_sig: Option<bls12381::Signature>,
+}
+
+impl MultiSignature {
+    pub fn new(
+        validator_bitmask: Vec<bool>,
+        aggregated_signature: Option<bls12381::Signature>,
+    ) -> Self {
+        Self {
+            validator_bitmask,
+            multi_sig: aggregated_signature,
+        }
+    }
+
+    pub fn empty() -> Self {
+        Self::default()
```

Comment:
I'd remove the default impl and just hand write the empty one to make it more clear